src/collections/_documentation/platforms/javascript/sourcemaps/troubleshooting.md

```diff
@@ -19,36 +25,102 @@ Double-check that your deployed, final JavaScript files have `sourceMappingURL`
 
 Alternately, instead of `sourceMappingURL`, you can set a `SourceMap` HTTP header on your minified file. If this header is present, Sentry will use it to discover the location of your source map.
 
-### Verify artifact names match sourceMappingURL
+### Verify artifact names match `sourceMappingURL` value
 
-When [uploading source maps to Sentry](#uploading-source-maps-to-sentry), you must name your source map files with the same name found in `sourceMappingURL`.
+The `sourceMappingURL` comment on the last line of your bundled or minified JavaScript file tells Sentry (or the browser) where to locate the corresponding source map. This can either be a fully qualified URL, a relative path, or the filename itself. When uploading artifacts to Sentry, you must name your source map files with the same name found in the  `sourceMappingURL` comment.
 
-For example, if you have the following in a minified application file, `app.min.js`:
+i. e. if your file is similar to:
 
 ```javascript
-//-- end app.min.js
-//# sourceMappingURL=https://example.com/dist/js/app.min.js.map
+    // -- end script.min.js
+    //# sourceMappingURL=script.min.js.map
 ```
 
-Sentry will look for a matching artifact named exactly `https://example.com/dist/js/app.min.js.map`.
+then your uploaded artifact should also be named `script.min.js.map`. 
 
-Note also that Sentry will resolve relative paths. For example, if you have the following:
+or if your file is similar to: 
 
-```JavaScript
-// -- end app.min.js (located at https://example.com/dist/js/app.min.js)
-//# sourceMappingURL=app.min.js.map
+```javascript
+    //-- end script.min.js
+    //# sourceMappingURL=https://example.com/dist/js/script.min.js.map
 ```
 
-Sentry will resolve `sourceMappingURL` relative to `https://example.com/dist/js/` (the root path from which `app.min.js` was served). You will again need to name your source map with the full URL: `https://example.com/dist/js/app.min.js.map`.
+then your uploaded artifact should also be named `https://example.com/dist/js/script.min.js.map`
+
+
+### Verify artifact names match stack trace frames
+
+If you’ve uploaded source maps and they aren’t applying to your code in an issue in Sentry, take a look at the filename in the stack trace frames, for example, `/scripts/script.min.js`. You can also open up the JSON of the event and look for the `abs_path` to see exactly where we’re attempting to resolve the file  - i.e. `http://localhost:8000/scripts/script.js`. The uploaded artifact names must match one of these values.
```

Comment:
> You can also open up the JSON of the event and look for the `abs_path` to see exactly where we’re attempting to resolve the file

Maybe we should encourage them to _only_ look at the JSON view (if the values are different)? We might also want to clarify where the JSON view is – many people miss it.